File: gin_microservice/rabbitmq/rabbitmq.go
package rabbitmq

import (
	"github.com/streadway/amqp"
)

func Connect() (*amqp.Channel, error) {
	conn, err := amqp.Dial("amqp://guest:guest@localhost:5672/")
	if err != nil {
		return nil, err
	}

	return conn.Channel()
}

File: actix_microservice/src/rabbitmq.rs
// use lapin::{Connection, ConnectionProperties};
// use tokio_amqp::LapinTokioExt;

// pub async fn connect() -> Connection {
//     let addr = "amqp://guest:guest@localhost:5672//";
//     Connection::connect(&addr, ConnectionProperties::default().with_tokio()).await.unwrap()
// }


use lapin::{Connection, ConnectionProperties};
use tokio_amqp::LapinTokioExt;

pub async fn connect() -> Connection {
    let addr = "amqp://guest:guest@localhost:5672//";
    Connection::connect(&addr, ConnectionProperties::default().with_tokio()).await.expect("Failed to connect to RabbitMQ")
}
